### PR TITLE
Conformance tests

### PR DESCRIPTION
**Pull request by fingolfin**
This contains PR #1878 and PR #1881 plus one extra commit.

Closes #1881

### CODE REVIEW

**Comment by fingolfin** [comment]
This requires https://github.com/Nemocas/AbstractAlgebra.jl/pull/1827 which I mistakenly thought had been included in the latest AA release already, but it not even merged. So for now this PR here will fail.

File: test/flint/gfp-test.jl [hunk 1]
```diff
@@ -1,11 +1,8 @@
-function test_elem(R::Nemo.fpField)
-  return rand(R)
-end
-
 @testset "gfp.conformance_tests" begin
   # TODO: make this work with test_Field_interface_recursive
   for p in [13, next_prime(2^8), next_prime(2^16), next_prime(2^32)]
     test_Field_interface(Native.GF(p))
+    #test_Field_interface_recursive(Native.GF(p))
```
Comment (lgoettgens) [inline test/flint/gfp-test.jl:5]: Can you add a comment here why this fails?

**Review by lgoettgens — APPROVED** [review]

**Review by joschmitt — APPROVED** [review]

File: test/flint/gfp-test.jl (hunk 1 above)
Comment (fingolfin) [reply test/flint/gfp-test.jl:5]: I'll do it in a follow-up so we don't have to re-run CI again, OK?

Comment (lgoettgens) [reply test/flint/gfp-test.jl:5]: Fine

**Pull request merged**